### DISCUSSION TITLE
FileNotFoundError when storing media files in a folder instead of the database

### DISCUSSION THREAD

**Discussion by mathieumd** [sheltermanager/asm3#1559] [Q&A / Help] [answered]
When the conf is set to store files into a folder, media upload fails and Apache logs shows:
```
FileNotFoundError: [Errno 2] No such file or directory: '/srv/www/asm3_data//srv/www/asm3_db/asm3.sqlite/10.jpg', referer: https://gestion-chenil.example.com/animal_media?id=1&newmedia=1
```

`/etc/asm3.conf`:
```conf
db_type = SQLITE
db_name = /srv/www/asm3_db/asm3.sqlite
dbfs_store = file
dbfs_filestorage_folder = /srv/www/asm3_data
```

I guess the concatenation of `DBFS_FILESTORAGE_FOLDER` and `self.dbo.database` in file [/usr/lib/sheltermanager3/asm3/dbfs.py](https://github.com/sheltermanager/asm3/blob/master/src/asm3/dbfs.py#L111) is not working in this case, as it's trying to access the file `/srv/www/asm3_data//srv/www/asm3_db/asm3.sqlite/10.jpg`, which is `dbfs_filestorage_folder` + `db_name` + file).

Is anybody has a working folder storage?

**Comment by bobintetley** [comment, marked as answer]
Hi, thanks for raising this. The file storage class was there before SQLite support and we never used them together. I've committed a fix to the codebase now to stop this being an issue in future (if you can pull from head you will get the fix). Unfortunately, the only workaround for now is to not use the file storage type with SQLite until you have the fix.

**Comment by mathieumd** [comment]
Thanks Robin.
I'm using the `.deb` release, currently at version 48, so I'm gonna have to wait to get the fix :)
As you said, until then I have to use `dbfs_store = database`. By the way, when I will change it back to `dbfs_store = file`, I guess it will not purge the database of binaries files already uploaded, right? So, is it safe and enough to remove the `base64:` entries from the table `dbfs`?

**Reply by bobintetley** [reply]
once you switch to dbfs_store = file, you can run the cron.py script with the parameter "maint_switch_dbfs_storage". It will move all of your base64 encoded files out of the database and put them in the file storage for you.
